src/main/java/jenkins/scm/api/SCMRevisionAction.java

```diff
@@ -44,11 +54,36 @@
     /**
      * Constructor.
      *
-     * @param rev the {@link SCMRevision}.
+     * @param revision the {@link SCMRevision}.
+     * @deprecated use {@link #SCMRevisionAction(SCMSource, SCMRevision)}
+     */
+    @Restricted(DoNotUse.class)
```

Comment:
`@Restricted` on top of `@Deprecated` is just superfluous. Deprecation itself suffices for the intended purpose. Restrictions should be limited to newly introduced members which for technical reasons must be `public`.